src/coreclr/nativeaot/Runtime/Full/CMakeLists.txt

```diff
@@ -57,14 +61,17 @@ add_custom_command(
 
 set_source_files_properties("${CMAKE_CURRENT_BINARY_DIR}/AsmOffsets.inc" PROPERTIES GENERATED TRUE)
 
+# Runtime and Runtime.ServerGC share AsmOffsets.inc and assembler helpers (for Windows ARM/ARM64).
+# Avoid race condition during a build using a separate target.
 add_custom_target(
-  AsmOffsets
-  DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/AsmOffsets.inc"
+  RuntimeAsmHelpers
+  DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/AsmOffsets.inc" "${RUNTIME_SOURCES_ARCH_ASM_RESOLVED}"
 )
 
-# The AsmOffsets.inc will be built as a dependency of the Runtime
-add_dependencies(Runtime AsmOffsets)
-add_dependencies(Runtime.ServerGC AsmOffsets)
+add_dependencies(Runtime RuntimeAsmHelpers)
+add_dependencies(Runtime.ServerGC RuntimeAsmHelpers)
+
+# REVIEW: COMPILE_DEFINITIONS_LIST is not defined here. Should we remove the following code?
```

Comment:
Does adding defines for the ASM work without this, or is it completely broken right now?

Either way, it sounds good to me to clean it up.

Reply:
We set all required preprocessor definitions here:
https://github.com/dotnet/runtimelab/blob/248949a1ecdfa34c89c2832b0c273c849e46ab55/src/coreclr/nativeaot/Runtime/CMakeLists.txt#L233

The code in this file is missing
```CMake
get_directory_property(COMPILE_DEFINITIONS_LIST COMPILE_DEFINITIONS)
```
However, even if we add that line, the code would still do nothing, because it does not handle complex conditions that we use, e.g., `$<$<OR:$<CONFIG:DEBUG>,$<CONFIG:CHECKED>>:DEBUG;_DEBUG;...>`.  I am going to remove it.